fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/ReduceCastRule.java

```diff
@@ -130,12 +126,17 @@ public boolean checkCastTypeReduceAble(Type parent, Type child, Type grandChild)
         return childCompatibleType != Type.INVALID && parentCompatibleType != Type.INVALID;
     }
 
-    public ScalarOperator reduceDateToDatetimeCast(BinaryPredicateOperator operator) {
-        ScalarOperator child1 = operator.getChild(0);
-        ScalarOperator child2 = operator.getChild(1);
-        ScalarOperator castChild = child1.getChild(0);
-        LocalDateTime originalDateTime = ((ConstantOperator) child2).getDatetime();
-        LocalDateTime bottomDateTime = ((ConstantOperator) child2).getDatetime().toLocalDate().atTime(0, 0, 0, 0);
+    private ScalarOperator reduceDateToDatetimeCast(BinaryPredicateOperator operator) {
+        ScalarOperator castChild = operator.getChild(0).getChild(0);
+        ConstantOperator child2 = (ConstantOperator) operator.getChild(1);
+
+        if (child2.isNull()) {
+            return operator;
```

Comment:
Does returning null here mean giving up optimization? If yes, then why can't you modify the type of null and then proceed with the optimization

Reply:
Null meanings the child is non-value and the rule depend on the value, it's should optimize by other rule, not here